src/middlewares/cors.rs

```diff
@@ -0,0 +1,51 @@
+use std::pin::Pin;
+
+use futures::{FutureExt, TryFutureExt, future};
+use gotham::handler::HandlerFuture;
+use gotham::hyper::StatusCode;
+use gotham::hyper::header::{HeaderMap, ORIGIN};
+use gotham::middleware::Middleware;
+use gotham::prelude::*;
+use gotham::state::State;
+
+use std::env;
+
+#[derive(Clone, NewMiddleware)]
+pub struct CorsMiddleware;
+
+impl Middleware for CorsMiddleware {
+    fn call<Chain>(self, state: State, chain: Chain) -> Pin<Box<HandlerFuture>>
+        where
+            Chain: FnOnce(State) -> Pin<Box<HandlerFuture>>,
+        {
+            let allowed_origins = env::var("WHITELIST").unwrap_or_else(|_| "*".to_string());
+
+            let header_map = HeaderMap::borrow_from(&state);
+
+            let requests_origin = match header_map.get(ORIGIN) {
+                Some(origin) => origin.to_str().unwrap().to_string(),
+                None => "".to_string(),
+            };
+
+            let response_allowed_origin = if allowed_origins.contains(',') {
+                allowed_origins.split(',').find(|origin| origin.trim() == requests_origin).unwrap_or("").to_owned()
+            } else {
+                allowed_origins
+            };
+
+            let result = chain(state);
```

Comment:
Will this be reused later?

Reply:
Also, `chain` is not clear to an outsider, can you change this name to something more informative?

Reply:
> Will this be reused later?

I think it will not reused later

Reply:
> Also, `chain` is not clear to an outsider, can you change this name to something more informative?

This is the gotham description about what [chain](https://docs.rs/gotham/latest/gotham/middleware/chain/index.html) does:

> Defines the types for connecting multiple middleware into a “chain” when forming a pipeline.

I'm not sure what another name could have, any suggestions?